src/sqlfluff/rules/ambiguous/AM08.py

```diff
@@ -0,0 +1,107 @@
+"""Implementation of Rule AM08."""
+
+from typing import Optional, Tuple
+
+from sqlfluff.core.parser import (
+    ComparisonOperatorSegment,
+    KeywordSegment,
+    LiteralSegment,
+    WhitespaceSegment,
+)
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_AM08(BaseRule):
+    """Join clauses should be present.
+
+    **Anti-pattern**
+
+    Cross joins are valid, but rare in the wild - and more often created by mistake than on purpose.
+    This rule catches situations where a cross join has been specified,
+    but not explicitly and so the risk of a mistaken cross join is highly likely.
+
```

Comment:
I think we should probably address people using the `WHERE` clause a join condition here. I think it's an anti-pattern, so it should still be a failure, but there's a question on what to do with it:

```sql
SELECT *
FROM bar
JOIN baz
WHERE bar.x = baz.x
AND bar.z > 5
```

This is functionally like having an `ON` clause, but the condition has been put in a `WHERE` clause instead. This would be what someone does in the case of "comma style" join too e.g.

```sql
SELECT *
FROM bar, baz
WHERE bar.x = baz.x
AND bar.z > 5
```

(I agree with you to ignore queries like this for now - and good to see you've added a test case for it 💖)

What I'm not sure of here is the fix, if we bluntly turn it into a `CROSS JOIN`, then it's still functionally the same, but that's not really the _intent_ of the rule. A better fix would be to extract the relevant conditions from the `WHERE` and put them in an `ON` clause instead, so that the first query would get fixed to:

```sql
SELECT *
FROM bar
JOIN baz
ON bar.x = baz.x
WHERE bar.z > 5
```

How hard do you think that would be to add up front?

Reply:
Simplistic where clauses (single condition or multiple conditions with AND statements only) should be relatively easy.

Complex conditions (AND/OR combinations, bracketed expressions, subqueries) will be super tricky to always get right.

Let me try to come up with some test cases.
Best case - autofixer will be able to solve _some_ of cases.
Worst case - we will reuse these test cases for 'violation without auto fix'.